src/router.ts

```diff
@@ -264,7 +264,10 @@ export function createRouter(options: RouterOptions): Router {
       // for same reason TS thinks rawLocation.params can be undefined
       matcher.resolve(
         'params' in rawLocation
-          ? { ...rawLocation, params: encodeParams(rawLocation.params) }
+          ? {
+              ...rawLocation,
+              params: encodeParams(applyToParamsRaw(rawLocation.params)),
```

Comment:
A few thoughts for refactoring:

- `encodeParams` could accept numbers
- we could create a copy of `rawLocation` before L256, and add an else at L261 to change the `params` property to an encoded version without numbers, save the decoded version as a local variable `params` to use it later

Reply:
`encodeParams` is accepting `string | number` but the `bind` doesn't pick up correct overload, I will just create a const with the `normalized` params and use that across